File: python/autocomplete.py
words = ["apple", "apricot", "grape", "plum", "peach", "orange", "strawberry"]

def auto_complete(str):
  return [word for word in words if word[0:len(str)] == str.lower()]


print(auto_complete('ap'))
print(auto_complete('pl'))
print(auto_complete('za'))
print(auto_complete('Gr'))
print(auto_complete('app'))

File: js/autocomplete.js
const words = ["apple", "apricot", "grape", "plum", "peach", "orange", "strawberry"]

const autoComplete = str => words.filter(word => word.slice(0, str.length) === str.toLowerCase());

console.log(autoComplete('ap'));
console.log(autoComplete('pl'));
console.log(autoComplete('za'));
console.log(autoComplete('Gr'));
console.log(autoComplete('app'));
